lib/generators/deimos/schema_class_generator.rb

```diff
@@ -0,0 +1,231 @@
+      def _schema_base(schema=nil)
+        @schema_base ||= Deimos.schema_backend_class.new(schema: extract_schema(schema),
+                                                         namespace: extract_namespace(schema))
+      end
+
+      # Unload the schema
+      def _clear_schema_base!
+        @schema_base = nil
+      end
+
+      # Parses the schema in dot syntax from a given Schema Path.
+      # Handles different cases for File/Schema names.
+      # @return [String] The name of the schema in the format of com.my.namespace.MySchema
+      def _parse_schema_from_path(schema_path)
+        return schema_path unless schema_path =~ %r{/}
+
+        full_schema_path = File.absolute_path(schema_path)
+        schema_name = File.basename(full_schema_path, '.avsc')
+        namespace_dir = File.dirname(full_schema_path).
+          delete_prefix("#{_schema_path}/").
+          gsub('/', '.')
+
+        "#{namespace_dir}.#{schema_name}"
+      end
+
+      # Converts Avro::Schema::NamedSchema's to String form for generated YARD docs.
+      # Recursively handles the typing for Arrays, Maps and Unions.
+      # @param avro_schema [Avro::Schema::NamedSchema]
+      # @return [String] A string representation of the Type of this SchemaField
+      def _field_type(avro_schema)
+        case avro_schema.type_sym
+        when :string, :boolean
+          avro_schema.type_sym.to_s.titleize
+        when :int, :long
+          'Integer'
+        when :float, :double
+          'Float'
+        when :record, :enum
+          "Deimos::#{schema_classname(avro_schema)}"
+        when :array
+          arr_t = deimos_field_type(Deimos::SchemaField.new('n/a', avro_schema.items))
+          "Array<#{arr_t}>"
+        when :map
+          map_t = deimos_field_type(Deimos::SchemaField.new('n/a', avro_schema.values))
+          "Hash<String, #{map_t}>"
+        when :union
+          types = avro_schema.schemas.map do |t|
+            deimos_field_type(Deimos::SchemaField.new('n/a', t))
+          end
+          types.join(', ')
+        when :null
+          'nil'
+        end
+      end
+
+      # @param avro_schema [Avro::Schema::NamedSchema]
+      # @return [Symbol]
+      def _schema_base_type(avro_schema)
+        case avro_schema.type_sym
+        when :array
+          _schema_base_type(avro_schema.items)
+        when :map
+          _schema_base_type(avro_schema.values)
+        when :union
+          avro_schema.schemas.map(&method(:_schema_base_type)).
+            reject { |schema| schema.type_sym == :null }.first
+        else
+          avro_schema
+        end
+      end
+
+      # @param avro_schema [Avro::Schema::NamedSchema]
+      # @return [String]
+      def _field_initialize_formatting(avro_schema)
+        field_type = _field_type(avro_schema)
+        case avro_schema.type_sym
+        when :record
+          "#{field_type}.initialize_from_payload"
+        when :enum
+          "#{field_type}.new"
```

Comment:
same question above - either we don't need `initialize_from_payload` and we can just use `new`, or we could just define an `initialize_from_payload` method on the enum and dispense with this method.

Reply:
yeah, that would be useful